sql/signal_handler.cc

```diff
@@ -57,6 +57,10 @@ static inline void output_core_info()
   int fd;
   if ((len= readlink("/proc/self/cwd", buff, sizeof(buff))) >= 0)
   {
+    if (len < sizeof(buff))
```

Comment:
You will need to cast to unsigned `len`, since readlink is returning sszite_t. Found with gcc-9.
There is MDEV-21978 what we can use later.